src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RethrowToPreserveStackDetailsTests.cs

```diff
@@ -511,12 +502,39 @@ void CatchAndRethrowExplicitly()
         }
         catch (Exception e)
         {
+            DoThrow();
+
+            void DoThrow()
+            {
+                throw e;
+            }
+        }
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task CA2200_NoDiagnosticsForThrowCaughtExceptionInLocalMethodAfterReassignment()
```

Comment:
Considering having a test for top-level statements (not necessarily in this PR).